src/contracts/payloads/PermissionedPayloadsController.sol

```diff
@@ -0,0 +1,88 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.0;
+
+import {IPayloadsControllerCore} from './interfaces/IPayloadsControllerCore.sol';
+import {PayloadsControllerCore} from './PayloadsControllerCore.sol';
+import {PayloadsControllerUtils} from './PayloadsControllerUtils.sol';
+import {WithPayloadsManager} from './WithPayloadsManager.sol';
+import {IPermissionedPayloadsController} from './interfaces/IPermissionedPayloadsController.sol';
+
+contract PermissionedPayloadsController is
+  PayloadsControllerCore,
+  WithPayloadsManager,
+  IPermissionedPayloadsController
+{
+  function initialize(
+    address owner,
+    address guardian,
+    address initialPayloadsManager,
+    UpdateExecutorInput[] calldata executors
+  ) external {
+    initialize(owner, guardian, executors);
+    _updatePayloadsManager(initialPayloadsManager);
+  }
+
+  function EXPIRATION_DELAY()
+    public
+    pure
+    override(PayloadsControllerCore, IPayloadsControllerCore)
+    returns (uint40)
+  {
+    return 10 days;
+  }
+
+  function GRACE_PERIOD()
+    public
+    pure
+    override(PayloadsControllerCore, IPayloadsControllerCore)
```

Comment:
I don't think you have to override interface right? just with contract should work

Reply:
You do need it. As interface and implementation contract inherited on the same level